django/contrib/auth/base_user.py

```diff
@@ -137,3 +136,7 @@ def get_session_auth_hash(self):
         """
         key_salt = "django.contrib.auth.models.AbstractBaseUser.get_session_auth_hash"
         return salted_hmac(key_salt, self.password).hexdigest()
+
+    @classmethod
+    def normalize_username(cls, username):
+        return unicodedata.normalize('NFKC', force_text(username))
```

Comment:
if no `USERNAME_FIELD` is defined, `self.get_username()` return None as default, and `force_text` will return a 'None' string instead of empty string. But the question is like above, can `USERNAME_FIELD` is not defined